tests/baselines/reference/variadicTuples2.types

```diff
@@ -465,7 +465,7 @@ fn1([1, 'abc']);  // [number, string]
 fn1([1, 'abc', true]);  // [string, boolean]
 >fn1([1, 'abc', true]) : [string, boolean]
 >fn1 : <T, U>(t: [...unknown[], T, U]) => [T, U]
->[1, 'abc', true] : [number, string, boolean]
+>[1, 'abc', true] : [number, string, true]

```

Comment:
Following inference and instantiation, the contextual type for the array literal is `[...unknown[], string, boolean]`. The contextual type for the `true` literal is the is computed from `unknown | string | boolean` which previously would reduce to just `unknown`. With this PR we don't perform subtype reduction, so `boolean` survives in the contextual union type and causes the `true` literal to stay `true`. But the change doesn't really matter since we've already inferred `boolean` for `U`.